db/migrations/mysql/000054_upgrade_posts_v6.0.up.sql

```diff
@@ -0,0 +1,90 @@
+CREATE PROCEDURE MigrateRootId_Posts () BEGIN DECLARE ParentId_EXIST INT;
+SELECT COUNT(*)
+FROM INFORMATION_SCHEMA.COLUMNS
+WHERE TABLE_NAME = 'Posts'
+  AND table_schema = DATABASE()
+  AND COLUMN_NAME = 'ParentId' INTO ParentId_EXIST;
+IF(ParentId_EXIST > 0) THEN
+    UPDATE Posts SET RootId = ParentId WHERE RootId = '' AND RootId != ParentId;
+END IF;
+END;
+
+CALL MigrateRootId_Posts ();
+DROP PROCEDURE IF EXISTS MigrateRootId_Posts;
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'Posts'
+        AND table_schema = DATABASE()
+        AND column_name = 'ParentId'
+    ) > 0,
+    'ALTER TABLE Posts DROP COLUMN ParentId;',
```

Comment:
Just noting that this would become very inefficient as we would be doing 3 alter operation in sequence (3 full table rebuilds) instead of the single (combined) one we have now. Is there any way to retain the existing logic?

Reply:
Hmm.. It might be hard to make it idempotent and combine the operations.

Reply:
I am guessing it will become more verbose but should be doable (and probably worth it). It's a one-off case really.

Reply:
Yeah we can do that. My main idea was to have atomic operations to have more control over this but you are absolutely right from the table rebuild point of view.